### DISCUSSION TITLE
Coveo Headless - groupBy functionality

### DISCUSSION THREAD

**Discussion by marbor3** [coveo/ui-kit#2853] [Q&A]
Hello! 

I'm in the middle of quite some task - migrating from CoveoJSS for Sitecore into Headless, and I cannot find one important feature... Up to a point that searching through sources of `ui-kit` I cannot find it. I'm 95% sure it's a missing feature, so just wanted to make sure my thinking is correct, and ask if I can maybe help out. 

**I need to create a Tab component with a number of results inside each Tab.**
I guess what I need is `groupBy` https://docs.coveo.com/en/1453/build-a-search-ui/group-by-operations
At least this functionality is used in the JSS version uses that - we listen for some events there, hook into the query builder, and add some stuff. 

Some questions:
- is there groupBy functionality in Coveo Headless and should I read the manual more carefully? 
- or this is not yet implemented, I should either wait patiently or open a PR here - **that's my best guess and the reason I write here instead of asking ChatBot** 
- or I should use `Facets` for it, but then could a Facet accept an expression (as Tabs do)? Or maybe I should keep using `HeadlessTab` and just somehow return the count for each tab? 

Thanks!
Best,
Marcin

**Comment by louis-bompart** [comment]
Hey @marbor3 , sorry for the late answer!

Would https://docs.coveo.com/en/atomic/latest/reference/components/atomic-segmented-facet/ do the trick?

**Reply by marbor3** [reply]
Hey! 

Hmm, I was looking rather into `headless` as we have to heavily modify the styling. 
But no worries I was able to solve it with `preprocessRequest` and modifying request there. Maybe not the most elegant way, but did the trick!
